adjusted the project code and configuration to run in the new enigmampc/docker-environment

reuvenpo opened the pull request:
This PR adds support for running Salad inside enimgampc/docker-environment.
Please check out the PR at enigmampc/docker-environment#19 to see what i'm doing with all these changes. I'm still going to polish some stuff here, but any input is obviously welcome :)
I definitely intend on making sure that i didn't break the frontend service, but i haven't done that yet.

reuvenpo committed:
adjusted the project code and configuration to run in the new enigmampc/docker-environment

reuvenpo committed:
removed unused test files

These files are leftovers from previous work, and have no place
here anymore. We will write new tests in another location.

reuvenpo committed:
small fixes and cleanups

reuvenpo committed:
updated discovery-cli to 0.1.7 and removed the build/ directory completely.

This is done after enigmampc/discovery-cli#19 was fixed and closed.

reuvenpo committed:
fixed typo in client/src/enigmaTokenSmartContract.js

reuvenpo committed:
This commit is a refactor of the 02 migration. Read full description.

* refactored and simplified the 02 migration slightly
* added a loop that waits for a worker to be available in the Enigma network

lacabra commented on package.json:
```diff
@@ -6,23 +6,32 @@
+    "stop": "yarn dc stop",
+    "compile": "yarn dc compile",
+    "test": "yarn dc compile && mocha --exit ./test/api.test.js",
+    "migrate": "yarn dc compile && yarn dc migrate",
```
The latest version of `discovery-cli` enforces that `compile` is always run before `migrate` so `yarn dc migrate` suffices here.

reuvenpo replied:
:+1:

lacabra commented on package.json:
```diff
@@ -6,23 +6,32 @@
     "test": "test"
   },
+  "devDependencies": {
+    "@enigmampc/discovery-cli": "0.1.7",
```
latest is now `0.1.8`

reuvenpo replied:
:+1:

reuvenpo committed:
updated discovery-cli version and simplified `yarn migrate`

reuvenpo committed:
changed the way that the enigma smart contract addresses are collected

Previously, when running in the discovery-cli environment, the ethereum
addresses of the Enigma and Enigma Token contracts were just collected
from the JSON files of their ABI, and when running in the new
docker-environment, they were collected by fetching from an http server
that publishes them inside that network.

Now, the addresses are expected to be found in environment variables on
the server running the operator, and the client in turn queries the
operator for their values.

reuvenpo committed:
WIP on integration tests for salad, to be run under the docker-environment

reuvenpo committed:
adjusted frontend code to allow overriding the operator address

fredfortier commented on frontend/src/components/App.js:
```diff
@@ -42,8 +40,8 @@ class App extends Component {
             this.props.initializeWeb3(web3);
-            // TODO: Bootstrap the operator url during build
-            const salad = new CoinjoinClient( undefined, web3 );
+            const protocolHost = window.location.protocol + '//' + window.location.hostname;
```
We should define an endpoint for the operator api (e.g. http://host/api). The frontend should be in the domain root. A reverse proxy (the nginx service) should rewrite the api route on port 80 to the operator internal endpoint.

reuvenpo replied:
I don't really know how to set up and configure nginx. do you think you could do it as a PR to this branch? (or, just a commit on this branch?)

fredfortier replied:
I think that it can be a separate PR. We just need to keep track of it. I can take care of it.

fredfortier commented:
Looks good expect for the one conversation above.

reuvenpo committed:
simplified some parts of the salas SeC and removed lots of allocations from verify_signature

reuvenpo committed:
small aesthetic change

reuvenpo committed:
Merge remote-tracking branch 'origin/develop' into adjust-to-docker-environment

fredfortier committed:
DEV: Conditionally reading Enigma addresses from JSON files during deployment and propagating to other components + minor logging fixes

lacabra commented:
@reuvenpo In order to run this properly in the new `docker-environment` you need to push new vresions of the `docker-compose.hw.yml` and `docker-compose.sw.yml`, which in the case of `docker-environment` are these two files:

- https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.yml
- https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.hw.yml

As right now it's using the older network, and things are not working properly.

reuvenpo committed:
ammendment to previous commit by fred

reuvenpo commented:
> @reuvenpo In order to run this properly in the new `docker-environment` you need to push new vresions of the `docker-compose.hw.yml` and `docker-compose.sw.yml`, which in the case of `docker-environment` are these two files:
> 
>     * https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.yml
> 
>     * https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.hw.yml
> 
> 
> As right now it's using the older network, and things are not working properly.

as discussed privately, this is irrelevant.

reuvenpo committed:
Merge remote-tracking branch 'origin/develop' into adjust-to-docker-environment

fredfortier committed:
FIX: Fixed issue with broadcasting quorum and other updates

reuvenpo committed:
Combined the two env vars describing the mongo address and port into one describing the url

reuvenpo committed:
small modification to .dockerignore

fredfortier commented on package.json:
```diff
@@ -6,23 +6,33 @@
+    "start": "yarn dc start",
+    "stop": "yarn dc stop",
+    "compile": "yarn dc compile",
+    "test": "yarn dc compile && mocha --exit ./test/api.test.js",
```
Why did you change the workflow here? `migrate` is the right command, not compile. If you need a different workflow for system tests, I would recommend creating a new command.

fredfortier committed:
FIX: Fixed breaking changes related to unit testing

fredfortier committed:
Merge remote-tracking branch 'origin/adjust-to-docker-environment' into adjust-to-docker-environment

fredfortier committed:
FIX: Fixed more breaking changes. All tests passing.

fredfortier committed:
FIX: Fixing breaking change to the CI

fredfortier committed:
DEV: Troubleshooting an issue with one of the tests in the CI

fredfortier committed:
DEV: Troubleshooting an issue with one of the tests in the CI
